File: ObjC/ObjC/HCMSession.m
//
//  HCMSession.m
//  ObjC
//
//  Created by harry on 2/27/15.
//  Copyright (c) 2015 Harry Cheung. All rights reserved.
//

#import "HCMSession.h"

@implementation HCMSession

- (id)initWithTrack:(HCMTrack *)track
          startTime:(double)startTime {
  self = [super init];
  if (self) {
    _track = track;
    _startTime = startTime;
    _laps = [[NSMutableArray alloc] init];
    _bestLap = nil;
  }
  return self;
}

- (id)initWithTrack:(HCMTrack *)track {
  return [self initWithTrack:track startTime:[[NSDate date] timeIntervalSince1970]];
}

- (void)dealloc {
  [_laps release];
  [super dealloc];
}

@end

File: Shared/CPP/HCMSession.cpp
//
//  HCMSession.m
//  CPP
//
//  Copyright (c) 2015 Harry Cheung. All rights reserved.
//

#include "HCMSession.h"

HCMSession::HCMSession(HCMTrack *track, double startTime) {
  this->track = track;
  this->startTime = startTime;
  this->duration = 0;
  this->bestLap = NULL;
}

HCMSession::~HCMSession() {
  for (std::vector<HCMLap *>::iterator it = laps.begin(); it != laps.end(); ++it) {
    delete *it;
  }
}